code/aind_ccf_reg/preprocess.py

```diff
@@ -0,0 +1,296 @@
+        arr_mask = self._cleanup_mask(arr_mask)
+
+        # convert numpy array to ants image
+        ants_img_mask = ants.from_numpy(
+                                arr_mask.astype("float32"), 
+                                spacing=self.ants_img.spacing,
+                                origin=self.ants_img.origin, 
+                                direction=self.ants_img.direction)
+
+        return ants_img_mask
+
+
+class Preprocess():
+    """
+    Class to Preprocess lightsheet data
+    1. resample to isotropic to have same resolution of the SPIM template
+    2. N4 bias correction
+    3. intensity normalization 
+    """
+    def __init__(self, args, input_data, reference_data):
+        self.args = args
+        self.input_data = input_data
+        self.reference_data = reference_data
+            
+    def resample(self, ants_img, ants_template):    
+        """ Resample OMEZarr image to the resolution of template """
+        logger.info(f"Resample OMEZarr image to the resolution of template")
+        ants_img = ants.resample_image(
+            ants_img, ants_template.spacing, False, 1
+        )
+
+        logger.info(f"Resampled OMEZarr dataset: {ants_img}")
+                
+        # #------------- TODO: do we need? -------------#
+
+        # logger.info(f"Size of resampled image: {ants_img.shape}")
+
+        # # convert input data to tiff into reference voxel resolution
+        # downsampled_file_path = Path(
+        #     f"{self.args['metadata_folder']}/{self.args['downsampled_file']}"
+        # )
+        # ants.image_write(ants_img, str(downsampled_file_path))
+
+        # # convert data to uint16
+        # im = io.imread(str(downsampled_file_path)).astype(np.uint16)
+        # downsampled16bit_file_path = Path(
+        #     f"{self.args['metadata_folder']}/{self.args['downsampled16bit_file']}"
+        # )
+        # tifffile.imwrite(str(downsampled16bit_file_path), im)
+        # #------------- TODO end -------------#
+
+        write_and_plot_image(
+            ants_img,
+            data_path=self.args["prep_params"].get("resample_path"),
+            plot_path=self.args["prep_params"].get("resample_figpath"), 
+            vmin=0, vmax=500)
+               
+        return ants_img
+    
+    
+    def compute_mask(self, ants_img):
+        """ compute make """
+        logger.info("Computing Mask")
+        
+        start_time = datetime.now()
+        mask = Masking(ants_img)
+        ants_img_mask = mask.run()
+        end_time = datetime.now()
+
+        logger.info(
+            f"Mask Complete, execution time: {end_time - start_time} s -- image {ants_img_mask}"
+        )
+    
+        write_and_plot_image(
+            ants_img_mask,
+            data_path=self.args["prep_params"].get("mask_path"), 
+            plot_path=self.args["prep_params"].get("mask_figpath"), 
+            vmin=0, vmax=1)
+
+        return ants_img_mask    
+
+    
+    def compute_N4(self, ants_img, ants_img_mask):
```

Comment:
Consider to have an extra parameter for n4_bias_params. This could be None and if so, then you could use the default params you have there. This will give us flexibility in the future.